subprojects/launcher/src/main/java/org/gradle/tooling/internal/provider/action/BuildActionSerializer.java

```diff
@@ -342,13 +342,17 @@ public ClientProvidedPhasedAction read(Decoder decoder) throws Exception {
         final Set<InternalJvmTestRequest> internalJvmTestRequests;
         final InternalDebugOptions debugOptions;
         final Map<String, List<InternalJvmTestRequest>> taskAndTests;
+        final boolean isRunDefaultTasks;
+        final List<String> tasks;
 
-        public TestExecutionRequestPayload(Set<InternalTestDescriptor> testDescriptors, Set<String> classNames, Set<InternalJvmTestRequest> internalJvmTestRequests, InternalDebugOptions debugOptions, Map<String, List<InternalJvmTestRequest>> taskAndTests) {
+        public TestExecutionRequestPayload(Set<InternalTestDescriptor> testDescriptors, Set<String> classNames, Set<InternalJvmTestRequest> internalJvmTestRequests, InternalDebugOptions debugOptions, Map<String, List<InternalJvmTestRequest>> taskAndTests, boolean isRunDefaultTasks, List<String> tasks) {
             this.testDescriptors = testDescriptors;
             this.classNames = classNames;
             this.internalJvmTestRequests = internalJvmTestRequests;
             this.debugOptions = debugOptions;
             this.taskAndTests = taskAndTests;
+            this.isRunDefaultTasks = isRunDefaultTasks;
+            this.tasks = tasks;
```

Comment:
@sterling Maybe we should encapsulate these fields into a new `InternalTaskRequest` class. WDYT?